messaging/src/main/java/org/axonframework/eventhandling/ReplayToken.java

```diff
@@ -97,55 +188,39 @@ public static boolean isReplay(Message<?> message) {
      */
     public static boolean isReplay(TrackingToken trackingToken) {
         return WrappedToken.unwrap(trackingToken, ReplayToken.class)
-                .map(rt -> rt.isReplay())
-                .orElse(false);
-
+                           .map(rt -> rt.isReplay())
+                           .orElse(false);
     }
 
     /**
-     * Return the relative position at which a reset was triggered for this Segment.
-     * In case a replay finished or no replay is active, an {@code OptionalLong.empty()} will be returned.
-     *
-     * @return the relative position at which a reset was triggered for this Segment
-     */
-    public static OptionalLong getTokenAtReset(TrackingToken trackingToken) {
-        return WrappedToken.unwrap(trackingToken, ReplayToken.class)
-                           .map(rt -> rt.getTokenAtReset().position())
-                .orElse(OptionalLong.empty());
-    }
-
-    /**
-     * Initialize a ReplayToken, using the given {@code tokenAtReset} to represent the position at which a reset was
-     * triggered. The current token is reset to the initial position.
+     * Extracts the context from a {@code trackingToken} of the matching {@code contextClass}.
      * <p>
-     * Using the {@link #createReplayToken(TrackingToken)} is preferred, as it covers cases where a replay is started
+     * Will resolve to null if the provided token is not a {@link ReplayToken} or the context does not match the
+     * provided {@code contextClass}.
      *
-     * @param tokenAtReset The token representing the position at which the reset was triggered.
+     * @param trackingToken The tracking token to extract the context from
+     * @param contextClass  The class the context should match
+     * @param <T>           The type of the context
+     * @return The context, if matching
      */
-    public ReplayToken(TrackingToken tokenAtReset) {
-        this(tokenAtReset, null);
+    public static <T> T replayContext(TrackingToken trackingToken, @Nonnull Class<T> contextClass) {
+        return WrappedToken.unwrap(trackingToken, ReplayToken.class)
+                           .map(ReplayToken::context)
+                           .filter(c -> c.getClass().isAssignableFrom(contextClass))
+                           .map(contextClass::cast)
+                           .orElse(null);
     }
 
     /**
-     * Initializes a ReplayToken with {@code tokenAtReset} which represents the position at which a reset was triggered
-     * and the {@code newRedeliveryToken} which represents current token.
+     * Return the relative position at which a reset was triggered for this Segment. In case a replay finished or no
+     * replay is active, an {@code OptionalLong.empty()} will be returned.
      *
-     * @param tokenAtReset       The token representing the position at which the reset was triggered
-     * @param newRedeliveryToken The current token
+     * @return the relative position at which a reset was triggered for this Segment
```

Comment:
The reference to Segments doesn't make sense here. The concept of Segments is beyond the scope of Tokens.

Reply:
It's been there for 2.5 years, but I adjusted it